Update EIP-8141: add support for atomic batching (#11395)
ethereum/EIPs: EIPS/eip-8141.md

Frame executes as regular call where the caller address is `sender`. This mode effectively acts on behalf of the transaction sender and can only be used after explicitly approved.

##### `SENDER_ATOMIC` Mode

> Same as `SENDER` mode, except that when a `SENDER_ATOMIC` frame reverts, all consecutive `SENDER_ATOMIC` frames that immediately precede the frame are also reverted, and all consecutive `SENDER_ATOMIC` frames that immediately follow the frame are skipped from execution.

Reviewer: I think that most developers would expect atomic to be the default setting

With frame transactions, EOA wallets today can reap the key benefit of AA -- gas abstraction, including sending sponsored transactions, paying gas in ERC-20 tokens, and more.

### Atomicity

By default, frames are not atomic.  This is important because we don't want SENDER frames to revert VERIFY frames, which would revert gas payments.

> One might wonder why we didn't make SENDER frames atomic with one another.  The reason is that there are use cases such as Example 2 where the paymaster (using a VERIFY frame) needs the guarantee that the following SENDER frame does not revert no matter what happens to the following SENDER frames, since the paymaster would lose money if ERC-20 tokens were not transferred to it.

Reviewer: "following" wording is a bit unclear here

```suggestion
One might wonder why we didn't make SENDER frames atomic with one another.  The reason is that there are use cases such as Example 2 where the paymaster (using a VERIFY frame) needs the guarantee that the following SENDER frame (which transfers ERC-20 tokens to the paymaster) does not revert no matter what happens to the subsequent SENDER frames.
```
